src/backtracking/knight_tour.rs

```diff
@@ -0,0 +1,219 @@
+//! This module contains the implementation of the Knight's Tour problem.
+//!
+//! The Knight's Tour is a classic chess problem where the objective is to move a knight to every square on a chessboard exactly once.
+
+/// Finds the Knight's Tour starting from the specified position.
+///
+/// # Arguments
+///
+/// * `size_x` - The width of the chessboard.
+/// * `size_y` - The height of the chessboard.
+/// * `start_x` - The x-coordinate of the starting position.
+/// * `start_y` - The y-coordinate of the starting position.
+///
+/// # Returns
+///
+/// A tuple containing a boolean indicating whether a tour was found and the tour matrix if found.
+pub fn find_knight_tour(
+    size_x: usize,
+    size_y: usize,
+    start_x: usize,
+    start_y: usize,
+) -> (bool, Vec<Vec<usize>>) {
```

Comment:
```suggestion
) -> Option(Vec<Vec<usize>>) {
```

When the solution does not exist one can return `None`.

Reply:
Done.